[82] update project README (#88)
RonasIT/laravel-chat: README.md

Reviewer: let's remove Laravel 5.5 point as we use minimal 12 version

Reviewer: Let's move this description to the API endpoints block

Reviewer: let's describe how to pass routes as arguments and add description about automatically routes disabling in case some routes type are passed as args

```

3. Set your project's User model to the `chat.classes.user_model` config.

## Integration with [LaravelSwagger](https://github.com/RonasIT/laravel-swagger)

This package includes an OpenAPI documentation file. To include it in your project's documentation, you need to register it in the `auto-doc.additional_paths` config:

`vendor/ronasit/laravel-chat/documentation.json`

## API Endpoints

All routes are registered by default, you can change the route registration by calling `Route::chat()` in your routes file (e.g. `routes/api.php`).
- feel free to call `Route::chat()` helper inside any route wrappers like `group`, `prefix`, etc. to wrap package routes;
- calling `Route::chat()` without args will add all package routes inside the calling helper place;
- calling `Route::chat()` with `ChatRouteActionEnum` cases as arguments will register **only** the specified routes — all others are automatically disabled:

```php
use RonasIT\Chat\Enums\ChatRouteActionEnum;

Route::prefix('api')->group(function () {
    Route::chat(
        ChatRouteActionEnum::ConversationsSearch,
        ChatRouteActionEnum::MessageCreate,
    );
});
```

> All endpoints require authentication. The routes are protected by the `auth` middleware.

Author: Fixed

php artisan vendor:publish --provider=RonasIT\\Chat\\ChatServiceProvider
```

> 3. Set your project's User model to the `chat.classes.user_model` config.

Reviewer: ```suggestion
3. If you use non default `App\Models\User` model - update `chat.classes.user_model` config.
```

## Integration with [LaravelSwagger](https://github.com/RonasIT/laravel-swagger)

This package includes an OpenAPI documentation file. To include it in your project's documentation, you need to register it in the `auto-doc.additional_paths` config:

`vendor/ronasit/laravel-chat/documentation.json`

## API Endpoints

All routes are registered by default, you can change the route registration by calling `Route::chat()` in your routes file (e.g. `routes/api.php`).

> - feel free to call `Route::chat()` helper inside any route wrappers like `group`, `prefix`, etc. to wrap package routes;

Reviewer: ```suggestion
- feel free to call `Route::chat()` helper inside any route wrappers like `group`, `prefix`, etc. to wrap package routes; **In this case default package's routes start to return 404.**
```

- calling `Route::chat()` without args will add all package routes inside the calling helper place;
- calling `Route::chat()` with `ChatRouteActionEnum` cases as arguments will register **only** the specified routes — all others are automatically disabled:

```php
// routes/api.php
use RonasIT\Chat\Enums\ChatRouteActionEnum;

Route::middleware('auth')->group(function () {
    Route::chat(
        ChatRouteActionEnum::ConversationsSearch,
        ChatRouteActionEnum::MessageCreate,
    );

> });

Reviewer: ```suggestion
Route::prefix('api')->group(function () {
    Route::middleware('auth_group')->group(function () {
        Route::chat(
            ChatRouteActionEnum::ConversationsSearch,
            ChatRouteActionEnum::MessageCreate,
        );
    });
});
```

```

> All endpoints require authentication. The routes are protected by the `auth` middleware.

Reviewer: ```suggestion
All endpoints are using the current auth user context, so it critical to always wrap them into the `auth` middleware.
```

`vendor/ronasit/laravel-chat/documentation.json`

## API Endpoints

All endpoints require authentication. The routes are protected by the `auth` middleware.

### Conversations

> | Method | URL | Action | Description |

Reviewer: not sure we need to add the `action` column

```suggestion
| Method | URL  | Description |
```

|--------|-----|--------|-------------|
| `GET` | `/conversations` | `conversations_search` | List all conversations the authenticated user is a member of. |
| `GET` | `/conversations/{id}` | `conversation_get` | Retrieve a single conversation by its ID. |
| `DELETE` | `/conversations/{id}` | `conversation_delete` | Delete a conversation. |
| `GET` | `/users/{userId}/conversation` | `conversation_get_by_user` | Get the private conversation between the authenticated user and the specified user. |

### Messages

| Method | URL | Action | Description |
|--------|-----|--------|-------------|

> | `GET` | `/messages` | `messages_search` | List messages for a given conversation. |

Reviewer: ```suggestion
| `GET` | `/messages` | `messages_search` | List of messages related to the current user's conversations. |
```

| `POST` | `/messages` | `message_create` | Create a new message. |

> | `POST` | `/messages/{id}/read-to` | `messages_read` | Mark all messages up to the specified message ID as read. |

Reviewer: ```suggestion
| `POST` | `/messages/{id}/read-to` | `messages_read` | Mark all messages in the target message's conversation as read  up to the specified message ID. |
```

| `POST` | `/messages/{id}/pin` | `message_pin` | Pin a message to its conversation. |
| `POST` | `/messages/{id}/unpin` | `message_unpin` | Unpin a message from its conversation. |

## Broadcast Events

The package uses Laravel's broadcasting system to notify conversation members in real time. All events are delivered over private channels named `chat.{userId}`, where `{userId}` is the ID of the recipient.

### Channel Authorization

> A user may only listen on their own private channel `chat.{userId}`. The channel is authorized when the authenticated user's ID matches `{userId}`.

Reviewer: ```suggestion
A user may listen on only his own private channel `chat.{userId}`. The channel is authorized when the authenticated user's ID matches `{userId}`.
```

### Events

#### `conversation.created`

Sent to all conversation members **except the creator** when a new conversation is created.

This event is triggered when:
- A new group conversation is created.
- A private conversation is created implicitly (e.g., when a user sends the first message to another user via `recipient_id`).

**Payload:**

```jsonc
{
    "id": 1,
    "type": "private|group",
    "title": "string or null",
    "last_updated_at": "2024-01-01T00:00:00.000000Z",
    "created_at": "2024-01-01T00:00:00.000000Z",

>     "last_message": { /* MessageResource */ }

Author: I believe this is a useless note. This will look like unnecessary visual clutter.

}
```

---

#### `conversation.updated`

Sent to **all conversation members** when a conversation is modified.

This event is triggered when:
- A conversation's properties (e.g., title, cover) are updated.

> - A message is pinned or unpinned in the conversation.

Author: This may be useful; clarification added.

A new message is created (the conversation's `last_updated_at` is updated as a side effect).
